src/dwarf/resolver.rs

```diff
@@ -264,6 +309,42 @@ mod tests {
 
     use test_log::test;
 
+    fn mksym(name: &'static str, addr: Addr) -> DWSymInfo<'static> {
+        DWSymInfo {
+            name,
+            addr,
+            size: 0,
+            sym_type: SymType::Function,
+        }
+    }
+
+    fn mksyms(syms: &[(&'static str, Addr)]) -> DebugSyms<'static> {
+        let syms = syms.iter().map(|(name, addr)| mksym(name, *addr)).collect();
+        DebugSyms::new(syms)
+    }
+
+    /// Check that our `DebugSyms` type allows for proper lookup by name.
+    #[test]
+    fn debug_symbol_by_name_lookup() {
+        let syms = [];
+        let syms = mksyms(&syms);
+        assert_eq!(syms.find_by_name("").count(), 0);
+        assert_eq!(syms.find_by_name("foobar").count(), 0);
+
+        let syms = [("foobar", 0x123), ("foobar", 0x126), ("bar", 0x127)];
+        let syms = mksyms(&syms);
+        assert_eq!(syms.find_by_name("").count(), 0);
+
+        let mut found = syms.find_by_name("foobar").collect::<Vec<_>>();
+        let () = found.sort_by_key(|sym| sym.addr);
```

Comment:
why not define that syms are sorted by name, and then by addr from lowest to highest. This doesn't add any overhead, but gives more defined order within duplicates (and we can even do binary search within equal symbols in the future)?

Reply:
Yeah that makes sense. Changed.